### PR TITLE
Add command to validate configuration

### PR DESCRIPTION
**Pull request by joshwget**
Adds a new `ros config validate` command that uses most of the code I wrote for the [online validator](https://github.com/joshwget/validator). The schema is still pretty vague in many places, but after this PR we'd be able to slowly add fields and improve it.

#617

### CODE REVIEW

File: scripts/schema.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
+{
+  "type": "object",
+  "additionalProperties": false,
```
Comment (SvenDowideit) [inline scripts/schema.json]: what is this file used for? could we remove it, and have something print out the `schema` constant in config/schema.go? (or even better, have the structure deserialization do things?)

yes, maybe this suggestion is for a later PR - but having the same text twice doesn't feel like a good idea.

Comment (joshwget) [inline scripts/schema.json]: `config/schema.go` is generated based on this, so I'm not really sure that counts as having the same text twice. It's not strictly necessary, but I think it's good practice to have the actual schema file around.

**Comment by SvenDowideit** [comment]
LGTM - @deniseschannon ?

**Comment by deniseschannon** [comment]
LGTM for docs - but not merging cause it's not green. @joshwget

**Comment by SvenDowideit** [comment]
ya for green :)

**Pull request merged**